File: backend/app/services/baerenklau_beat.py
import re

from app.schemas.script import ScriptBeat

_BAERENKLAU_RE = re.compile(r"bärenklau|baerenklau|bärenklauer|baerenklauer", re.IGNORECASE)


def is_baerenklau_beat(beat: ScriptBeat) -> bool:
    haystack = f"{beat.scene_title or ''} {beat.text[:400]}"
    return bool(_BAERENKLAU_RE.search(haystack))


def find_baerenklau_beats(beats: list[ScriptBeat]) -> list[ScriptBeat]:
    return [beat for beat in beats if is_baerenklau_beat(beat)]


def resolve_part1_beats(beats: list[ScriptBeat]) -> list[ScriptBeat]:
    """Teil 1: alle Beats (typisch ein Gesamttext-Beat)."""
    return list(beats)


def resolve_part1_beat(beats: list[ScriptBeat]) -> ScriptBeat | None:
    resolved = resolve_part1_beats(beats)
    return resolved[0] if resolved else None


def first_baerenklau_beat(beats: list[ScriptBeat]) -> ScriptBeat | None:
    found = find_baerenklau_beats(beats)
    return found[0] if found else None

File: frontend/lib/show/baerenklauBeat.ts
import type { ScriptBeat } from "@/lib/types/script";

const BAERENKLAU_RE = /bärenklau|baerenklau|bärenklauer|baerenklauer/i;

export function isBaerenklauBeat(beat: ScriptBeat): boolean {
  const haystack = `${beat.scene_title ?? ""} ${beat.text.slice(0, 400)}`;
  return BAERENKLAU_RE.test(haystack);
}

export function findBaerenklauBeats(beats: ScriptBeat[]): ScriptBeat[] {
  return beats.filter(isBaerenklauBeat);
}

export function part1Beats(beats: ScriptBeat[]): ScriptBeat[] {
  return beats;
}

export function isPart1Beat(beat: ScriptBeat, beats: ScriptBeat[]): boolean {
  const part1 = part1Beats(beats);
  return part1.some((b) => b.id === beat.id);
}
